### PR TITLE
Appropriate changes to vacation requests and tests

### LINKED ISSUE #75: Ability to draft vacation requests

### PR DESCRIPTION
**Pull request by Iiro127**

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/timebank/api/persistence/repositories/VacationsRequestsRepository.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ class VacationsRequestsRepository: AbstractRepository<VacationRequest, UUID>() {
      * @param after LocalDate to retrieve requests after given date
      * @return List of VacationRequests
      */
-    suspend fun listVacationRequest(personId: UUID?, before: LocalDate?, after: LocalDate?): List<VacationRequest> {
+    suspend fun listVacationRequest(personId: UUID?, before: LocalDate?, after: LocalDate?, draft: Boolean?): List<VacationRequest> {
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/timebank/api/persistence/repositories/VacationsRequestsRepository.kt:23]: This seems to be missing concrete usage of this new parameter.

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestsTest.kt [hunk 2]
```diff
@@ -42,6 +42,7 @@ class VacationRequestsTest: AbstractTest() {
     days = 2,
     type = VacationType.VACATION,
     message = "Lomaa!!!",
+    draft = false,
```
Comment (villejuutila) [inline src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestsTest.kt:45]: Add tests for filtering listing with the `draft` parameter.

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/timebank/api/persistence/repositories/VacationsRequestsRepository.kt (hunk 1 above)
Comment (Iiro127) [reply src/main/kotlin/fi/metatavu/timebank/api/persistence/repositories/VacationsRequestsRepository.kt:23]: Resolved by implementing usage

File: src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestsTest.kt (hunk 2 above)
Comment (Iiro127) [reply src/test/kotlin/fi/metatavu/timebank/api/test/functional/tests/VacationRequestsTest.kt:45]: Resolved

**Review by villejuutila — APPROVED** [review]

**Pull request merged**